proto/substrait/algebra.proto

```diff
@@ -230,6 +230,78 @@ message AggregateRel {
   }
 }
 
+// ConsistentPartitionWindowRel provides the ability to perform calculations across sets of rows
+// that are related to the current query row. It can be used to execute window functions where
+// all the windows share the same partitioning and ordering.
+message ConsistentPartitionWindowRel {
+  RelCommon common = 1;
+  Rel input = 2;
+  repeated WindowRelFunction window_functions = 3;
+  repeated Expression partition_expressions = 4;
+  repeated SortField sorts = 5;
+
+  substrait.extensions.AdvancedExtension advanced_extension = 10;
+
+  // A WindowFunction can define itself WindowSpec(sorts columns and partitions keys)
+  // But WindowRelFunction remove the WindowSpec property, because all the WindowRelFunctions
+  // share the same WindowSpec() in WindowRel.
```

Comment:
```suggestion
  // This message mirrors the `WindowFunction` message but removes the fields defining the partition,
  // sorts, and bounds, since those must be consistent across the various functions in this rel.  Refer
  // to the `WindowFunction` message for a description of these fields.
```